### PR TITLE
Modularizing app

### PR DESCRIPTION
**Pull request by yte0**

### CODE REVIEW

**Comment by arnfinn** [comment]
Great:)

File: NAMESPACE [hunk 1]
```diff
@@ -16,6 +16,12 @@ export(indicator_rows)
 export(load_data)
 export(mod_quality_overview_server)
 export(mod_quality_overview_ui)
+export(mod_sidebar_qo_server)
+export(mod_sidebar_qo_ui)
+export(mod_table_legend_server)
+export(mod_table_legend_ui)
+export(mod_top_navbar_server)
+export(mod_top_navbar_ui)
```
Comment (arnfinn) [inline NAMESPACE]: Is it OK to skip the `mod_` prefix, or do you think we need it?

**Comment by arnfinn** [comment]
@yte0: Are you looking at the broken tests, or should I do it?

File: NAMESPACE (hunk 1 above)
Comment (yte0) [reply NAMESPACE]: It can be skip:)

**Comment by yte0** [comment]
was planning to look into issue #37 Dropdown in table. but if you have other think to do i can do the testing first and then look at issue #37 later!

**Comment by arnfinn** [comment]
Please do #37:)

I can fix tests.

File: R/mod_quality_overview.R [hunk 2]
```diff
@@ -94,242 +50,89 @@ mod_quality_overview_ui <- function(id) {
...
+  app_data <- app_data(config)
+  register_data <- app_data[["register_data"]]
+  grouped_by_hf <- app_data[["grouped_by_hf"]]
+  grouped_by_rhf <- app_data[["grouped_by_rhf"]]
+  grouped_by_hospital <- app_data[["grouped_by_hospital"]]
+  national_data <- app_data[["national_data"]]
+
+  selected_units <- shiny::callModule(
+    top_navbar_server,
+    NULL,
```
Comment (arnfinn) [inline R/mod_quality_overview.R]: Why `NULL` here? I thought `input, output, session` were implicit and you only had to send `app_data` and `config`.

File: R/mod_quality_overview.R [hunk 3]
```diff
@@ -94,242 +50,89 @@ mod_quality_overview_ui <- function(id) {
...
+    config = config
+  )
+  filter_indicator <- shiny::reactiveValues()
+  filter_indicator$level <- shiny::callModule(
+    table_legend_server,
+    NULL
+  )
+  filter_indicator$indicator <- shiny::callModule(
+    sidebar_qo_server,
+    NULL,
```
Comment (arnfinn) [inline R/mod_quality_overview.R]: As above

File: R/mod_quality_overview.R [hunk 4]
```diff
@@ -94,242 +50,89 @@ mod_quality_overview_ui <- function(id) {
...
+  selected_units <- shiny::callModule(
+    top_navbar_server,
+    NULL,
+    app_data = app_data,
+    config = config
+  )
+  filter_indicator <- shiny::reactiveValues()
+  filter_indicator$level <- shiny::callModule(
+    table_legend_server,
+    NULL
```
Comment (arnfinn) [inline R/mod_quality_overview.R]: As above

**Comment by arnfinn** [comment]
I will also try to make unit tests for the new modules.

File: R/mod_quality_overview.R (hunk 4 above)
Comment (yte0) [reply R/mod_quality_overview.R]: my bad, it was not the best solution. I have corrected it in the last commit. Good job reviewing:-)

**Comment by arnfinn** [comment]
In commit de7fb311 the file `quality_overview` was created, but the file `mod_quality_overview` was not deleted. OK if I rewrite history from the commit before de7fb311 until now?

**Comment by arnfinn** [comment]
I have now rewritten history... The easiest way to continue working on this branch is to locally delete it first. Thus:
```sh
git checkout master
git branch -D Modularizing_app # delete the local branch
git fetch # update
git checkout Modularizing_app # checkout fresh branch
```
Last commit should be 3945d06.

Commit 2e728be has not been incorporated yet, but I am afraid it will break the code.

**Pull request merged**